orchestrator/vendor/broadcaster/broadcaster/_base.py

```diff
@@ -0,0 +1,117 @@
+import asyncio
+from contextlib import asynccontextmanager
+from typing import Any, AsyncGenerator, AsyncIterator, Dict, Optional
+from urllib.parse import urlparse
+
+
+class Event:
+    def __init__(self, channel: str, message: str) -> None:
+        self.channel = channel
+        self.message = message
+
+    def __eq__(self, other: object) -> bool:
+        return (
+            isinstance(other, Event)
+            and self.channel == other.channel
+            and self.message == other.message
+        )
+
+    def __repr__(self) -> str:
+        return f"Event(channel={self.channel!r}, message={self.message!r})"
+
+
+class Unsubscribed(Exception):
+    pass
+
+
+class Broadcast:
```

Comment:
You could have used a submodule or only add this file, since no changes are needed in other files :3

Reply:
I left the broadcaster code as-is in case there are bugs with this commit, then it's easy to go to a different commit.

Yeah I considered a git submodule, but that would mean other orchestrator-core users will need to learn about that as well. And they're not the easiest thing in the world to work with.

Reply:
Note that this is not meant to be permanent, created #580 for a long term solution